### PR TITLE
Add 2.1 changelog

### PR DESCRIPTION
**Pull request by pohly**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:

It prepares for a v2.1.0 release.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/assign @msau42

### CODE REVIEW

**Comment by pohly** [comment]
@msau42, can you merge and tag it as v2.1.0?

File: CHANGELOG-2.0.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Changelog since v2.0.1
```
Comment (msau42) [inline CHANGELOG-2.0.md]: We are currently keeping a different file per minor version.  Looks like we missed a changelog file for 2.0 itself.

Comment (pohly) [reply CHANGELOG-2.0.md]: I've renamed the file.

We didn't have good changelog entries in the PRs leading up to v2.0.1, that probably explains it.

**Pull request merged**